### PR TITLE
Test Sparse Dataset Operations

### LINKED ISSUE #240: Test (and fix) sparse data support in dataset
The work in #237 has support for sparse dimensions in the rewrite of `class Dataset`. However, the testing is only rudimentary so far, and it is very likely that a number of things are currently broken.

### PR DESCRIPTION
**Pull request by OwenArnold**
Fixes #240.

Majority of the work involved adding tests as described in issue.

### CODE REVIEW

File: core/test/dataset_test.cpp [hunk 1]
```diff
@@ -249,6 +251,59 @@ TEST(DatasetTest, const_iterators_return_types) {
...
+
+TEST(DatasetTest, simple_sparse_slice) {
+
+  Dataset dataset;
+  auto var = makeVariable<double>({{Dim::Y, Dim::X}, {2, Dimensions::Sparse}});
+  var.sparseValues<double>()[0] = {4, 5, 6};
+  var.sparseValues<double>()[1] = {7, 8, 9};
+  dataset.setData("data", var);
+  dataset.setCoord(Dim::Y, makeVariable<double>({Dim::Y, 2}, {1, 2}));
+  auto sparseCoords = makeVariable<double>({Dim::X, Dimensions::Sparse});
```
[the remark this answers is not in the document]
Comment (OwenArnold) [reply core/test/dataset_test.cpp]: Currently there is no way to do this, I would need to spend some time establishing what the problem is. I've updated the comment with some pseudo code

[the remark this answers is not in the document]
Comment (OwenArnold) [reply core/test/dataset_test.cpp]: Maybe I should look into this some more, the `coords` method has a comment deliberately forbidding the return. https://github.com/scipp/scipp/blob/master/core/dataset.cpp#L310

[the remark this answers is not in the document]
Comment (nvaytet) [reply core/test/dataset_test.cpp]: Hi @OwenArnold, did you want to look into this more for this PR or delay this to another PR? Should i just merge this one?

[the remark this answers is not in the document]
Comment (OwenArnold) [reply core/test/dataset_test.cpp]: @nvaytet I think this needs to be looked at as a separate issue.

[the remark this answers is not in the document]
Comment (nvaytet) [reply core/test/dataset_test.cpp]: ok if you fix the conflicts then I will merge

**Pull request merged**